File: Parallel-Java/src/parallel/linq101/Generators.java
package parallel.linq101;

import org.javatuples.Pair;

import java.util.ArrayList;
import java.util.Collections;
import java.util.stream.Collectors;
import java.util.stream.IntStream;

public class Generators {
    public int rangeOfIntegers() {
        var numbers = IntStream.range(100, 100 + 50).boxed().
                map(n -> Pair.with(n, n % 2 == 1 ? "odd" : "even")).collect(Collectors.toList());
        for (var n : numbers)
            System.out.println("The number " + n.getValue0() + " is " + n.getValue1() + ".");
        return 0;
    }

    public int repeatNumber() {
        var numbers = new ArrayList<>(Collections.nCopies(10, 7));
        for(var n : numbers)
            System.out.println(n);
        return 0;
    }
}

File: Parallel-Csharp/src/Parallel/Linq101/Generators.cs
using System;
using System.Linq;

namespace Parallel.Linq101;
    public class Generators
    {
        public int RangeOfIntegers()
        {
            var numbers = Enumerable.Range(100, 50).Select(n => Tuple.Create(n, n % 2 == 1 ? "odd" : "even")).ToList();


            foreach (var n in numbers)
                Console.WriteLine($"The number {n.Item1} is {n.Item2}.");
            return 0;
        }

        public int RepeatNumber()
        {
            var numbers = Enumerable.Repeat(7, 10).ToList();

            foreach (var n in numbers)
                Console.WriteLine(n);
            
            return 0;
        }
    }

